// reactjs-nextjs-carousel/dist/index.es.js
import me, { forwardRef as Ft, useRef as Lt, useDebugValue as ut, createElement as An, useContext as jn, useState as In, useEffect as lt } from "react";
function $n(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Or = { exports: {} }, Be = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ft;
function Dn() {
  if (ft)
    return Be;
  ft = 1;
  var e = me, r = Symbol.for("react.element"), t = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, a = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function s(f, l, g) {
    var p, y = {}, b = null, w = null;
    g !== void 0 && (b = "" + g), l.key !== void 0 && (b = "" + l.key), l.ref !== void 0 && (w = l.ref);
    for (p in l)
      n.call(l, p) && !i.hasOwnProperty(p) && (y[p] = l[p]);
    if (f && f.defaultProps)
      for (p in l = f.defaultProps, l)
        y[p] === void 0 && (y[p] = l[p]);
    return { $$typeof: r, type: f, key: b, ref: w, props: y, _owner: a.current };
  }
  return Be.Fragment = t, Be.jsx = s, Be.jsxs = s, Be;
}
var Ue = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var dt;
function Nn() {
  return dt || (dt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = me, r = Symbol.for("react.element"), t = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), a = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), s = Symbol.for("react.provider"), f = Symbol.for("react.context"), l = Symbol.for("react.forward_ref"), g = Symbol.for("react.suspense"), p = Symbol.for("react.suspense_list"), y = Symbol.for("react.memo"), b = Symbol.for("react.lazy"), w = Symbol.for("react.offscreen"), j = Symbol.iterator, _ = "@@iterator";
    function W(o) {
      if (o === null || typeof o != "object")
        return null;
      var c = j && o[j] || o[_];
      return typeof c == "function" ? c : null;
    }
    var S = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function E(o) {
      {
        for (var c = arguments.length, v = new Array(c > 1 ? c - 1 : 0), O = 1; O < c; O++)
          v[O - 1] = arguments[O];
        T("error", o, v);
      }
    }
    function T(o, c, v) {
      {
        var O = S.ReactDebugCurrentFrame, V = O.getStackAddendum();
        V !== "" && (c += "%s", v = v.concat([V]));
        var G = v.map(function(z) {
          return String(z);
        });
        G.unshift("Warning: " + c), Function.prototype.apply.call(console[o], console, G);
      }
    }
    var P = !1, d = !1, I = !1, k = !1, ue = !1, le;
    le = Symbol.for("react.module.reference");
    function Re(o) {
      return !!(typeof o == "string" || typeof o == "function" || o === n || o === i || ue || o === a || o === g || o === p || k || o === w || P || d || I || typeof o == "object" && o !== null && (o.$$typeof === b || o.$$typeof === y || o.$$typeof === s || o.$$typeof === f || o.$$typeof === l || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      o.$$typeof === le || o.getModuleId !== void 0));
    }
    function we(o, c, v) {
      var O = o.displayName;
      if (O)
        return O;
      var V = c.displayName || c.name || "";
      return V !== "" ? v + "(" + V + ")" : v;
    }
    function Se(o) {
      return o.displayName || "Context";
    }
    function re(o) {
      if (o == null)
        return null;
      if (typeof o.tag == "number" && E("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof o == "function")
        return o.displayName || o.name || null;
      if (typeof o == "string")
        return o;
      switch (o) {
        case n:
          return "Fragment";
        case t:
          return "Portal";
        case i:
          return "Profiler";
        case a:
          return "StrictMode";
        case g:
          return "Suspense";
        case p:
          return "SuspenseList";
      }
      if (typeof o == "object")
        switch (o.$$typeof) {
          case f:
            var c = o;
            return Se(c) + ".Consumer";
          case s:
            var v = o;
            return Se(v._context) + ".Provider";
          case l:
            return we(o, o.render, "ForwardRef");
          case y:
            var O = o.displayName || null;
            return O !== null ? O : re(o.type) || "Memo";
          case b: {
            var V = o, G = V._payload, z = V._init;
            try {
              return re(z(G));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var te = Object.assign, ye = 0, oe, ie, ne, fe, u, h, D;
    function $() {
    }
    $.__reactDisabledLog = !0;
    function R() {
      {
        if (ye === 0) {
          oe = console.log, ie = console.info, ne = console.warn, fe = console.error, u = console.group, h = console.groupCollapsed, D = console.groupEnd;
          var o = {
            configurable: !0,
            enumerable: !0,
            value: $,
            writable: !0
          };
          Object.defineProperties(console, {
            info: o,
            log: o,
            warn: o,
            error: o,
            group: o,
            groupCollapsed: o,
            groupEnd: o
          });
        }
        ye++;
      }
    }
    function M() {
      {
        if (ye--, ye === 0) {
          var o = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: te({}, o, {
              value: oe
            }),
            info: te({}, o, {
              value: ie
            }),
            warn: te({}, o, {
              value: ne
            }),
            error: te({}, o, {
              value: fe
            }),
            group: te({}, o, {
              value: u
            }),
            groupCollapsed: te({}, o, {
              value: h
            }),
            groupEnd: te({}, o, {
              value: D
            })
          });
        }
        ye < 0 && E("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var C = S.ReactCurrentDispatcher, x;
    function A(o, c, v) {
      {
        if (x === void 0)
          try {
            throw Error();
          } catch (V) {
            var O = V.stack.trim().match(/\n( *(at )?)/);
            x = O && O[1] || "";
          }
        return `
` + x + o;
      }
    }
    var L = !1, N;
    {
      var H = typeof WeakMap == "function" ? WeakMap : Map;
      N = new H();
    }
    function m(o, c) {
      if (!o || L)
        return "";
      {
        var v = N.get(o);
        if (v !== void 0)
          return v;
      }
      var O;
      L = !0;
      var V = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var G;
      G = C.current, C.current = null, R();
      try {
        if (c) {
          var z = function() {
            throw Error();
          };
          if (Object.defineProperty(z.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(z, []);
            } catch (be) {
              O = be;
            }
            Reflect.construct(o, [], z);
          } else {
            try {
              z.call();
            } catch (be) {
              O = be;
            }
            o.call(z.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (be) {
            O = be;
          }
          o();
        }
      } catch (be) {
        if (be && O && typeof be.stack == "string") {
          for (var Y = be.stack.split(`
`), ae = O.stack.split(`
`), K = Y.length - 1, J = ae.length - 1; K >= 1 && J >= 0 && Y[K] !== ae[J]; )
            J--;
          for (; K >= 1 && J >= 0; K--, J--)
            if (Y[K] !== ae[J]) {
              if (K !== 1 || J !== 1)
                do
                  if (K--, J--, J < 0 || Y[K] !== ae[J]) {
                    var pe = `
` + Y[K].replace(" at new ", " at ");
                    return o.displayName && pe.includes("<anonymous>") && (pe = pe.replace("<anonymous>", o.displayName)), typeof o == "function" && N.set(o, pe), pe;
                  }
                while (K >= 1 && J >= 0);
              break;
            }
        }
      } finally {
        L = !1, C.current = G, M(), Error.prepareStackTrace = V;
      }
      var De = o ? o.displayName || o.name : "", ct = De ? A(De) : "";
      return typeof o == "function" && N.set(o, ct), ct;
    }
    function de(o, c, v) {
      return m(o, !1);
    }
    function Ie(o) {
      var c = o.prototype;
      return !!(c && c.isReactComponent);
    }
    function xe(o, c, v) {
      if (o == null)
        return "";
      if (typeof o == "function")
        return m(o, Ie(o));
      if (typeof o == "string")
        return A(o);
      switch (o) {
        case g:
          return A("Suspense");
        case p:
          return A("SuspenseList");
      }
      if (typeof o == "object")
        switch (o.$$typeof) {
          case l:
            return de(o.render);
          case y:
            return xe(o.type, c, v);
          case b: {
            var O = o, V = O._payload, G = O._init;
            try {
              return xe(G(V), c, v);
            } catch {
            }
          }
        }
      return "";
    }
    var Xe = Object.prototype.hasOwnProperty, Kr = {}, Jr = S.ReactDebugCurrentFrame;
    function Ke(o) {
      if (o) {
        var c = o._owner, v = xe(o.type, o._source, c ? c.type : null);
        Jr.setExtraStackFrame(v);
      } else
        Jr.setExtraStackFrame(null);
    }
    function fn(o, c, v, O, V) {
      {
        var G = Function.call.bind(Xe);
        for (var z in o)
          if (G(o, z)) {
            var Y = void 0;
            try {
              if (typeof o[z] != "function") {
                var ae = Error((O || "React class") + ": " + v + " type `" + z + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof o[z] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw ae.name = "Invariant Violation", ae;
              }
              Y = o[z](c, z, O, v, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (K) {
              Y = K;
            }
            Y && !(Y instanceof Error) && (Ke(V), E("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", O || "React class", v, z, typeof Y), Ke(null)), Y instanceof Error && !(Y.message in Kr) && (Kr[Y.message] = !0, Ke(V), E("Failed %s type: %s", v, Y.message), Ke(null));
          }
      }
    }
    var dn = Array.isArray;
    function dr(o) {
      return dn(o);
    }
    function pn(o) {
      {
        var c = typeof Symbol == "function" && Symbol.toStringTag, v = c && o[Symbol.toStringTag] || o.constructor.name || "Object";
        return v;
      }
    }
    function hn(o) {
      try {
        return Zr(o), !1;
      } catch {
        return !0;
      }
    }
    function Zr(o) {
      return "" + o;
    }
    function Qr(o) {
      if (hn(o))
        return E("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", pn(o)), Zr(o);
    }
    var qe = S.ReactCurrentOwner, yn = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, et, rt, pr;
    pr = {};
    function vn(o) {
      if (Xe.call(o, "ref")) {
        var c = Object.getOwnPropertyDescriptor(o, "ref").get;
        if (c && c.isReactWarning)
          return !1;
      }
      return o.ref !== void 0;
    }
    function mn(o) {
      if (Xe.call(o, "key")) {
        var c = Object.getOwnPropertyDescriptor(o, "key").get;
        if (c && c.isReactWarning)
          return !1;
      }
      return o.key !== void 0;
    }
    function gn(o, c) {
      if (typeof o.ref == "string" && qe.current && c && qe.current.stateNode !== c) {
        var v = re(qe.current.type);
        pr[v] || (E('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', re(qe.current.type), o.ref), pr[v] = !0);
      }
    }
    function bn(o, c) {
      {
        var v = function() {
          et || (et = !0, E("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", c));
        };
        v.isReactWarning = !0, Object.defineProperty(o, "key", {
          get: v,
          configurable: !0
        });
      }
    }
    function En(o, c) {
      {
        var v = function() {
          rt || (rt = !0, E("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", c));
        };
        v.isReactWarning = !0, Object.defineProperty(o, "ref", {
          get: v,
          configurable: !0
        });
      }
    }
    var wn = function(o, c, v, O, V, G, z) {
      var Y = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: r,
        // Built-in properties that belong on the element
        type: o,
        key: c,
        ref: v,
        props: z,
        // Record the component responsible for creating this element.
        _owner: G
      };
      return Y._store = {}, Object.defineProperty(Y._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(Y, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: O
      }), Object.defineProperty(Y, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: V
      }), Object.freeze && (Object.freeze(Y.props), Object.freeze(Y)), Y;
    };
    function Sn(o, c, v, O, V) {
      {
        var G, z = {}, Y = null, ae = null;
        v !== void 0 && (Qr(v), Y = "" + v), mn(c) && (Qr(c.key), Y = "" + c.key), vn(c) && (ae = c.ref, gn(c, V));
        for (G in c)
          Xe.call(c, G) && !yn.hasOwnProperty(G) && (z[G] = c[G]);
        if (o && o.defaultProps) {
          var K = o.defaultProps;
          for (G in K)
            z[G] === void 0 && (z[G] = K[G]);
        }
        if (Y || ae) {
          var J = typeof o == "function" ? o.displayName || o.name || "Unknown" : o;
          Y && bn(z, J), ae && En(z, J);
        }
        return wn(o, Y, ae, V, O, qe.current, z);
      }
    }
    var hr = S.ReactCurrentOwner, tt = S.ReactDebugCurrentFrame;
    function $e(o) {
      if (o) {
        var c = o._owner, v = xe(o.type, o._source, c ? c.type : null);
        tt.setExtraStackFrame(v);
      } else
        tt.setExtraStackFrame(null);
    }
    var yr;
    yr = !1;
    function vr(o) {
      return typeof o == "object" && o !== null && o.$$typeof === r;
    }
    function nt() {
      {
        if (hr.current) {
          var o = re(hr.current.type);
          if (o)
            return `

Check the render method of \`` + o + "`.";
        }
        return "";
      }
    }
    function _n(o) {
      {
        if (o !== void 0) {
          var c = o.fileName.replace(/^.*[\\\/]/, ""), v = o.lineNumber;
          return `

Check your code at ` + c + ":" + v + ".";
        }
        return "";
      }
    }
    var ot = {};
    function Tn(o) {
      {
        var c = nt();
        if (!c) {
          var v = typeof o == "string" ? o : o.displayName || o.name;
          v && (c = `

Check the top-level render call using <` + v + ">.");
        }
        return c;
      }
    }
    function at(o, c) {
      {
        if (!o._store || o._store.validated || o.key != null)
          return;
        o._store.validated = !0;
        var v = Tn(c);
        if (ot[v])
          return;
        ot[v] = !0;
        var O = "";
        o && o._owner && o._owner !== hr.current && (O = " It was passed a child from " + re(o._owner.type) + "."), $e(o), E('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', v, O), $e(null);
      }
    }
    function it(o, c) {
      {
        if (typeof o != "object")
          return;
        if (dr(o))
          for (var v = 0; v < o.length; v++) {
            var O = o[v];
            vr(O) && at(O, c);
          }
        else if (vr(o))
          o._store && (o._store.validated = !0);
        else if (o) {
          var V = W(o);
          if (typeof V == "function" && V !== o.entries)
            for (var G = V.call(o), z; !(z = G.next()).done; )
              vr(z.value) && at(z.value, c);
        }
      }
    }
    function Rn(o) {
      {
        var c = o.type;
        if (c == null || typeof c == "string")
          return;
        var v;
        if (typeof c == "function")
          v = c.propTypes;
        else if (typeof c == "object" && (c.$$typeof === l || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        c.$$typeof === y))
          v = c.propTypes;
        else
          return;
        if (v) {
          var O = re(c);
          fn(v, o.props, "prop", O, o);
        } else if (c.PropTypes !== void 0 && !yr) {
          yr = !0;
          var V = re(c);
          E("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", V || "Unknown");
        }
        typeof c.getDefaultProps == "function" && !c.getDefaultProps.isReactClassApproved && E("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function xn(o) {
      {
        for (var c = Object.keys(o.props), v = 0; v < c.length; v++) {
          var O = c[v];
          if (O !== "children" && O !== "key") {
            $e(o), E("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", O), $e(null);
            break;
          }
        }
        o.ref !== null && ($e(o), E("Invalid attribute `ref` supplied to `React.Fragment`."), $e(null));
      }
    }
    function st(o, c, v, O, V, G) {
      {
        var z = Re(o);
        if (!z) {
          var Y = "";
          (o === void 0 || typeof o == "object" && o !== null && Object.keys(o).length === 0) && (Y += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var ae = _n(V);
          ae ? Y += ae : Y += nt();
          var K;
          o === null ? K = "null" : dr(o) ? K = "array" : o !== void 0 && o.$$typeof === r ? (K = "<" + (re(o.type) || "Unknown") + " />", Y = " Did you accidentally export a JSX literal instead of a component?") : K = typeof o, E("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", K, Y);
        }
        var J = Sn(o, c, v, V, G);
        if (J == null)
          return J;
        if (z) {
          var pe = c.children;
          if (pe !== void 0)
            if (O)
              if (dr(pe)) {
                for (var De = 0; De < pe.length; De++)
                  it(pe[De], o);
                Object.freeze && Object.freeze(pe);
              } else
                E("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              it(pe, o);
        }
        return o === n ? xn(J) : Rn(J), J;
      }
    }
    function Cn(o, c, v) {
      return st(o, c, v, !0);
    }
    function On(o, c, v) {
      return st(o, c, v, !1);
    }
    var Pn = On, kn = Cn;
    Ue.Fragment = n, Ue.jsx = Pn, Ue.jsxs = kn;
  }()), Ue;
}
process.env.NODE_ENV === "production" ? Or.exports = Dn() : Or.exports = Nn();
var Z = Or.exports, Pr = { exports: {} }, Je = { exports: {} }, q = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var pt;
function Mn() {
  if (pt)
    return q;
  pt = 1;
  var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, a = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, f = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, g = e ? Symbol.for("react.concurrent_mode") : 60111, p = e ? Symbol.for("react.forward_ref") : 60112, y = e ? Symbol.for("react.suspense") : 60113, b = e ? Symbol.for("react.suspense_list") : 60120, w = e ? Symbol.for("react.memo") : 60115, j = e ? Symbol.for("react.lazy") : 60116, _ = e ? Symbol.for("react.block") : 60121, W = e ? Symbol.for("react.fundamental") : 60117, S = e ? Symbol.for("react.responder") : 60118, E = e ? Symbol.for("react.scope") : 60119;
  function T(d) {
    if (typeof d == "object" && d !== null) {
      var I = d.$$typeof;
      switch (I) {
        case r:
          switch (d = d.type, d) {
            case l:
            case g:
            case n:
            case i:
            case a:
            case y:
              return d;
            default:
              switch (d = d && d.$$typeof, d) {
                case f:
                case p:
                case j:
                case w:
                case s:
                  return d;
                default:
                  return I;
              }
          }
        case t:
          return I;
      }
    }
  }
  function P(d) {
    return T(d) === g;
  }
  return q.AsyncMode = l, q.ConcurrentMode = g, q.ContextConsumer = f, q.ContextProvider = s, q.Element = r, q.ForwardRef = p, q.Fragment = n, q.Lazy = j, q.Memo = w, q.Portal = t, q.Profiler = i, q.StrictMode = a, q.Suspense = y, q.isAsyncMode = function(d) {
    return P(d) || T(d) === l;
  }, q.isConcurrentMode = P, q.isContextConsumer = function(d) {
    return T(d) === f;
  }, q.isContextProvider = function(d) {
    return T(d) === s;
  }, q.isElement = function(d) {
    return typeof d == "object" && d !== null && d.$$typeof === r;
  }, q.isForwardRef = function(d) {
    return T(d) === p;
  }, q.isFragment = function(d) {
    return T(d) === n;
  }, q.isLazy = function(d) {
    return T(d) === j;
  }, q.isMemo = function(d) {
    return T(d) === w;
  }, q.isPortal = function(d) {
    return T(d) === t;
  }, q.isProfiler = function(d) {
    return T(d) === i;
  }, q.isStrictMode = function(d) {
    return T(d) === a;
  }, q.isSuspense = function(d) {
    return T(d) === y;
  }, q.isValidElementType = function(d) {
    return typeof d == "string" || typeof d == "function" || d === n || d === g || d === i || d === a || d === y || d === b || typeof d == "object" && d !== null && (d.$$typeof === j || d.$$typeof === w || d.$$typeof === s || d.$$typeof === f || d.$$typeof === p || d.$$typeof === W || d.$$typeof === S || d.$$typeof === E || d.$$typeof === _);
  }, q.typeOf = T, q;
}
var B = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ht;
function Fn() {
  return ht || (ht = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, a = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, f = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, g = e ? Symbol.for("react.concurrent_mode") : 60111, p = e ? Symbol.for("react.forward_ref") : 60112, y = e ? Symbol.for("react.suspense") : 60113, b = e ? Symbol.for("react.suspense_list") : 60120, w = e ? Symbol.for("react.memo") : 60115, j = e ? Symbol.for("react.lazy") : 60116, _ = e ? Symbol.for("react.block") : 60121, W = e ? Symbol.for("react.fundamental") : 60117, S = e ? Symbol.for("react.responder") : 60118, E = e ? Symbol.for("react.scope") : 60119;
    function T(m) {
      return typeof m == "string" || typeof m == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      m === n || m === g || m === i || m === a || m === y || m === b || typeof m == "object" && m !== null && (m.$$typeof === j || m.$$typeof === w || m.$$typeof === s || m.$$typeof === f || m.$$typeof === p || m.$$typeof === W || m.$$typeof === S || m.$$typeof === E || m.$$typeof === _);
    }
    function P(m) {
      if (typeof m == "object" && m !== null) {
        var de = m.$$typeof;
        switch (de) {
          case r:
            var Ie = m.type;
            switch (Ie) {
              case l:
              case g:
              case n:
              case i:
              case a:
              case y:
                return Ie;
              default:
                var xe = Ie && Ie.$$typeof;
                switch (xe) {
                  case f:
                  case p:
                  case j:
                  case w:
                  case s:
                    return xe;
                  default:
                    return de;
                }
            }
          case t:
            return de;
        }
      }
    }
    var d = l, I = g, k = f, ue = s, le = r, Re = p, we = n, Se = j, re = w, te = t, ye = i, oe = a, ie = y, ne = !1;
    function fe(m) {
      return ne || (ne = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), u(m) || P(m) === l;
    }
    function u(m) {
      return P(m) === g;
    }
    function h(m) {
      return P(m) === f;
    }
    function D(m) {
      return P(m) === s;
    }
    function $(m) {
      return typeof m == "object" && m !== null && m.$$typeof === r;
    }
    function R(m) {
      return P(m) === p;
    }
    function M(m) {
      return P(m) === n;
    }
    function C(m) {
      return P(m) === j;
    }
    function x(m) {
      return P(m) === w;
    }
    function A(m) {
      return P(m) === t;
    }
    function L(m) {
      return P(m) === i;
    }
    function N(m) {
      return P(m) === a;
    }
    function H(m) {
      return P(m) === y;
    }
    B.AsyncMode = d, B.ConcurrentMode = I, B.ContextConsumer = k, B.ContextProvider = ue, B.Element = le, B.ForwardRef = Re, B.Fragment = we, B.Lazy = Se, B.Memo = re, B.Portal = te, B.Profiler = ye, B.StrictMode = oe, B.Suspense = ie, B.isAsyncMode = fe, B.isConcurrentMode = u, B.isContextConsumer = h, B.isContextProvider = D, B.isElement = $, B.isForwardRef = R, B.isFragment = M, B.isLazy = C, B.isMemo = x, B.isPortal = A, B.isProfiler = L, B.isStrictMode = N, B.isSuspense = H, B.isValidElementType = T, B.typeOf = P;
  }()), B;
}
var yt;
function Yt() {
  return yt || (yt = 1, process.env.NODE_ENV === "production" ? Je.exports = Mn() : Je.exports = Fn()), Je.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var mr, vt;
function Ln() {
  if (vt)
    return mr;
  vt = 1;
  var e = Object.getOwnPropertySymbols, r = Object.prototype.hasOwnProperty, t = Object.prototype.propertyIsEnumerable;
  function n(i) {
    if (i == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(i);
  }
  function a() {
    try {
      if (!Object.assign)
        return !1;
      var i = new String("abc");
      if (i[5] = "de", Object.getOwnPropertyNames(i)[0] === "5")
        return !1;
      for (var s = {}, f = 0; f < 10; f++)
        s["_" + String.fromCharCode(f)] = f;
      var l = Object.getOwnPropertyNames(s).map(function(p) {
        return s[p];
      });
      if (l.join("") !== "0123456789")
        return !1;
      var g = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(p) {
        g[p] = p;
      }), Object.keys(Object.assign({}, g)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return mr = a() ? Object.assign : function(i, s) {
    for (var f, l = n(i), g, p = 1; p < arguments.length; p++) {
      f = Object(arguments[p]);
      for (var y in f)
        r.call(f, y) && (l[y] = f[y]);
      if (e) {
        g = e(f);
        for (var b = 0; b < g.length; b++)
          t.call(f, g[b]) && (l[g[b]] = f[g[b]]);
      }
    }
    return l;
  }, mr;
}
var gr, mt;
function zr() {
  if (mt)
    return gr;
  mt = 1;
  var e = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return gr = e, gr;
}
var br, gt;
function Wt() {
  return gt || (gt = 1, br = Function.call.bind(Object.prototype.hasOwnProperty)), br;
}
var Er, bt;
function Yn() {
  if (bt)
    return Er;
  bt = 1;
  var e = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var r = zr(), t = {}, n = Wt();
    e = function(i) {
      var s = "Warning: " + i;
      typeof console < "u" && console.error(s);
      try {
        throw new Error(s);
      } catch {
      }
    };
  }
  function a(i, s, f, l, g) {
    if (process.env.NODE_ENV !== "production") {
      for (var p in i)
        if (n(i, p)) {
          var y;
          try {
            if (typeof i[p] != "function") {
              var b = Error(
                (l || "React class") + ": " + f + " type `" + p + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof i[p] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw b.name = "Invariant Violation", b;
            }
            y = i[p](s, p, l, f, null, r);
          } catch (j) {
            y = j;
          }
          if (y && !(y instanceof Error) && e(
            (l || "React class") + ": type specification of " + f + " `" + p + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof y + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), y instanceof Error && !(y.message in t)) {
            t[y.message] = !0;
            var w = g ? g() : "";
            e(
              "Failed " + f + " type: " + y.message + (w ?? "")
            );
          }
        }
    }
  }
  return a.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (t = {});
  }, Er = a, Er;
}
var wr, Et;
function Wn() {
  if (Et)
    return wr;
  Et = 1;
  var e = Yt(), r = Ln(), t = zr(), n = Wt(), a = Yn(), i = function() {
  };
  process.env.NODE_ENV !== "production" && (i = function(f) {
    var l = "Warning: " + f;
    typeof console < "u" && console.error(l);
    try {
      throw new Error(l);
    } catch {
    }
  });
  function s() {
    return null;
  }
  return wr = function(f, l) {
    var g = typeof Symbol == "function" && Symbol.iterator, p = "@@iterator";
    function y(u) {
      var h = u && (g && u[g] || u[p]);
      if (typeof h == "function")
        return h;
    }
    var b = "<<anonymous>>", w = {
      array: S("array"),
      bigint: S("bigint"),
      bool: S("boolean"),
      func: S("function"),
      number: S("number"),
      object: S("object"),
      string: S("string"),
      symbol: S("symbol"),
      any: E(),
      arrayOf: T,
      element: P(),
      elementType: d(),
      instanceOf: I,
      node: Re(),
      objectOf: ue,
      oneOf: k,
      oneOfType: le,
      shape: Se,
      exact: re
    };
    function j(u, h) {
      return u === h ? u !== 0 || 1 / u === 1 / h : u !== u && h !== h;
    }
    function _(u, h) {
      this.message = u, this.data = h && typeof h == "object" ? h : {}, this.stack = "";
    }
    _.prototype = Error.prototype;
    function W(u) {
      if (process.env.NODE_ENV !== "production")
        var h = {}, D = 0;
      function $(M, C, x, A, L, N, H) {
        if (A = A || b, N = N || x, H !== t) {
          if (l) {
            var m = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw m.name = "Invariant Violation", m;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var de = A + ":" + x;
            !h[de] && // Avoid spamming the console because they are often not actionable except for lib authors
            D < 3 && (i(
              "You are manually calling a React.PropTypes validation function for the `" + N + "` prop on `" + A + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), h[de] = !0, D++);
          }
        }
        return C[x] == null ? M ? C[x] === null ? new _("The " + L + " `" + N + "` is marked as required " + ("in `" + A + "`, but its value is `null`.")) : new _("The " + L + " `" + N + "` is marked as required in " + ("`" + A + "`, but its value is `undefined`.")) : null : u(C, x, A, L, N);
      }
      var R = $.bind(null, !1);
      return R.isRequired = $.bind(null, !0), R;
    }
    function S(u) {
      function h(D, $, R, M, C, x) {
        var A = D[$], L = oe(A);
        if (L !== u) {
          var N = ie(A);
          return new _(
            "Invalid " + M + " `" + C + "` of type " + ("`" + N + "` supplied to `" + R + "`, expected ") + ("`" + u + "`."),
            { expectedType: u }
          );
        }
        return null;
      }
      return W(h);
    }
    function E() {
      return W(s);
    }
    function T(u) {
      function h(D, $, R, M, C) {
        if (typeof u != "function")
          return new _("Property `" + C + "` of component `" + R + "` has invalid PropType notation inside arrayOf.");
        var x = D[$];
        if (!Array.isArray(x)) {
          var A = oe(x);
          return new _("Invalid " + M + " `" + C + "` of type " + ("`" + A + "` supplied to `" + R + "`, expected an array."));
        }
        for (var L = 0; L < x.length; L++) {
          var N = u(x, L, R, M, C + "[" + L + "]", t);
          if (N instanceof Error)
            return N;
        }
        return null;
      }
      return W(h);
    }
    function P() {
      function u(h, D, $, R, M) {
        var C = h[D];
        if (!f(C)) {
          var x = oe(C);
          return new _("Invalid " + R + " `" + M + "` of type " + ("`" + x + "` supplied to `" + $ + "`, expected a single ReactElement."));
        }
        return null;
      }
      return W(u);
    }
    function d() {
      function u(h, D, $, R, M) {
        var C = h[D];
        if (!e.isValidElementType(C)) {
          var x = oe(C);
          return new _("Invalid " + R + " `" + M + "` of type " + ("`" + x + "` supplied to `" + $ + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return W(u);
    }
    function I(u) {
      function h(D, $, R, M, C) {
        if (!(D[$] instanceof u)) {
          var x = u.name || b, A = fe(D[$]);
          return new _("Invalid " + M + " `" + C + "` of type " + ("`" + A + "` supplied to `" + R + "`, expected ") + ("instance of `" + x + "`."));
        }
        return null;
      }
      return W(h);
    }
    function k(u) {
      if (!Array.isArray(u))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? i(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : i("Invalid argument supplied to oneOf, expected an array.")), s;
      function h(D, $, R, M, C) {
        for (var x = D[$], A = 0; A < u.length; A++)
          if (j(x, u[A]))
            return null;
        var L = JSON.stringify(u, function(H, m) {
          var de = ie(m);
          return de === "symbol" ? String(m) : m;
        });
        return new _("Invalid " + M + " `" + C + "` of value `" + String(x) + "` " + ("supplied to `" + R + "`, expected one of " + L + "."));
      }
      return W(h);
    }
    function ue(u) {
      function h(D, $, R, M, C) {
        if (typeof u != "function")
          return new _("Property `" + C + "` of component `" + R + "` has invalid PropType notation inside objectOf.");
        var x = D[$], A = oe(x);
        if (A !== "object")
          return new _("Invalid " + M + " `" + C + "` of type " + ("`" + A + "` supplied to `" + R + "`, expected an object."));
        for (var L in x)
          if (n(x, L)) {
            var N = u(x, L, R, M, C + "." + L, t);
            if (N instanceof Error)
              return N;
          }
        return null;
      }
      return W(h);
    }
    function le(u) {
      if (!Array.isArray(u))
        return process.env.NODE_ENV !== "production" && i("Invalid argument supplied to oneOfType, expected an instance of array."), s;
      for (var h = 0; h < u.length; h++) {
        var D = u[h];
        if (typeof D != "function")
          return i(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + ne(D) + " at index " + h + "."
          ), s;
      }
      function $(R, M, C, x, A) {
        for (var L = [], N = 0; N < u.length; N++) {
          var H = u[N], m = H(R, M, C, x, A, t);
          if (m == null)
            return null;
          m.data && n(m.data, "expectedType") && L.push(m.data.expectedType);
        }
        var de = L.length > 0 ? ", expected one of type [" + L.join(", ") + "]" : "";
        return new _("Invalid " + x + " `" + A + "` supplied to " + ("`" + C + "`" + de + "."));
      }
      return W($);
    }
    function Re() {
      function u(h, D, $, R, M) {
        return te(h[D]) ? null : new _("Invalid " + R + " `" + M + "` supplied to " + ("`" + $ + "`, expected a ReactNode."));
      }
      return W(u);
    }
    function we(u, h, D, $, R) {
      return new _(
        (u || "React class") + ": " + h + " type `" + D + "." + $ + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + R + "`."
      );
    }
    function Se(u) {
      function h(D, $, R, M, C) {
        var x = D[$], A = oe(x);
        if (A !== "object")
          return new _("Invalid " + M + " `" + C + "` of type `" + A + "` " + ("supplied to `" + R + "`, expected `object`."));
        for (var L in u) {
          var N = u[L];
          if (typeof N != "function")
            return we(R, M, C, L, ie(N));
          var H = N(x, L, R, M, C + "." + L, t);
          if (H)
            return H;
        }
        return null;
      }
      return W(h);
    }
    function re(u) {
      function h(D, $, R, M, C) {
        var x = D[$], A = oe(x);
        if (A !== "object")
          return new _("Invalid " + M + " `" + C + "` of type `" + A + "` " + ("supplied to `" + R + "`, expected `object`."));
        var L = r({}, D[$], u);
        for (var N in L) {
          var H = u[N];
          if (n(u, N) && typeof H != "function")
            return we(R, M, C, N, ie(H));
          if (!H)
            return new _(
              "Invalid " + M + " `" + C + "` key `" + N + "` supplied to `" + R + "`.\nBad object: " + JSON.stringify(D[$], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(u), null, "  ")
            );
          var m = H(x, N, R, M, C + "." + N, t);
          if (m)
            return m;
        }
        return null;
      }
      return W(h);
    }
    function te(u) {
      switch (typeof u) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !u;
        case "object":
          if (Array.isArray(u))
            return u.every(te);
          if (u === null || f(u))
            return !0;
          var h = y(u);
          if (h) {
            var D = h.call(u), $;
            if (h !== u.entries) {
              for (; !($ = D.next()).done; )
                if (!te($.value))
                  return !1;
            } else
              for (; !($ = D.next()).done; ) {
                var R = $.value;
                if (R && !te(R[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function ye(u, h) {
      return u === "symbol" ? !0 : h ? h["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && h instanceof Symbol : !1;
    }
    function oe(u) {
      var h = typeof u;
      return Array.isArray(u) ? "array" : u instanceof RegExp ? "object" : ye(h, u) ? "symbol" : h;
    }
    function ie(u) {
      if (typeof u > "u" || u === null)
        return "" + u;
      var h = oe(u);
      if (h === "object") {
        if (u instanceof Date)
          return "date";
        if (u instanceof RegExp)
          return "regexp";
      }
      return h;
    }
    function ne(u) {
      var h = ie(u);
      switch (h) {
        case "array":
        case "object":
          return "an " + h;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + h;
        default:
          return h;
      }
    }
    function fe(u) {
      return !u.constructor || !u.constructor.name ? b : u.constructor.name;
    }
    return w.checkPropTypes = a, w.resetWarningCache = a.resetWarningCache, w.PropTypes = w, w;
  }, wr;
}
var Sr, wt;
function zn() {
  if (wt)
    return Sr;
  wt = 1;
  var e = zr();
  function r() {
  }
  function t() {
  }
  return t.resetWarningCache = r, Sr = function() {
    function n(s, f, l, g, p, y) {
      if (y !== e) {
        var b = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw b.name = "Invariant Violation", b;
      }
    }
    n.isRequired = n;
    function a() {
      return n;
    }
    var i = {
      array: n,
      bigint: n,
      bool: n,
      func: n,
      number: n,
      object: n,
      string: n,
      symbol: n,
      any: n,
      arrayOf: a,
      element: n,
      elementType: n,
      instanceOf: a,
      node: n,
      objectOf: a,
      oneOf: a,
      oneOfType: a,
      shape: a,
      exact: a,
      checkPropTypes: t,
      resetWarningCache: r
    };
    return i.PropTypes = i, i;
  }, Sr;
}
if (process.env.NODE_ENV !== "production") {
  var Vn = Yt(), qn = !0;
  Pr.exports = Wn()(Vn.isElement, qn);
} else
  Pr.exports = zn()();
var Bn = Pr.exports;
const Te = /* @__PURE__ */ $n(Bn);
function kr() {
  return kr = Object.assign || function(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r];
      for (var n in t)
        Object.prototype.hasOwnProperty.call(t, n) && (e[n] = t[n]);
    }
    return e;
  }, kr.apply(this, arguments);
}
function Un(e, r) {
  if (e == null)
    return {};
  var t = Gn(e, r), n, a;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (a = 0; a < i.length; a++)
      n = i[a], !(r.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
  }
  return t;
}
function Gn(e, r) {
  if (e == null)
    return {};
  var t = {}, n = Object.keys(e), a, i;
  for (i = 0; i < n.length; i++)
    a = n[i], !(r.indexOf(a) >= 0) && (t[a] = e[a]);
  return t;
}
var Vr = Ft(function(e, r) {
  var t = e.color, n = t === void 0 ? "currentColor" : t, a = e.size, i = a === void 0 ? 24 : a, s = Un(e, ["color", "size"]);
  return /* @__PURE__ */ me.createElement("svg", kr({
    ref: r,
    xmlns: "http://www.w3.org/2000/svg",
    width: i,
    height: i,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: n,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, s), /* @__PURE__ */ me.createElement("polyline", {
    points: "15 18 9 12 15 6"
  }));
});
Vr.propTypes = {
  color: Te.string,
  size: Te.oneOfType([Te.string, Te.number])
};
Vr.displayName = "ChevronLeft";
const Hn = Vr;
function Ar() {
  return Ar = Object.assign || function(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r];
      for (var n in t)
        Object.prototype.hasOwnProperty.call(t, n) && (e[n] = t[n]);
    }
    return e;
  }, Ar.apply(this, arguments);
}
function Xn(e, r) {
  if (e == null)
    return {};
  var t = Kn(e, r), n, a;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (a = 0; a < i.length; a++)
      n = i[a], !(r.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
  }
  return t;
}
function Kn(e, r) {
  if (e == null)
    return {};
  var t = {}, n = Object.keys(e), a, i;
  for (i = 0; i < n.length; i++)
    a = n[i], !(r.indexOf(a) >= 0) && (t[a] = e[a]);
  return t;
}
var qr = Ft(function(e, r) {
  var t = e.color, n = t === void 0 ? "currentColor" : t, a = e.size, i = a === void 0 ? 24 : a, s = Xn(e, ["color", "size"]);
  return /* @__PURE__ */ me.createElement("svg", Ar({
    ref: r,
    xmlns: "http://www.w3.org/2000/svg",
    width: i,
    height: i,
    viewBox: "0 0 24 24",
    fill: "none",
    stroke: n,
    strokeWidth: "2",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }, s), /* @__PURE__ */ me.createElement("polyline", {
    points: "9 18 15 12 9 6"
  }));
});
qr.propTypes = {
  color: Te.string,
  size: Te.oneOfType([Te.string, Te.number])
};
qr.displayName = "ChevronRight";
const Jn = qr;
var se = function() {
  return se = Object.assign || function(r) {
    for (var t, n = 1, a = arguments.length; n < a; n++) {
      t = arguments[n];
      for (var i in t)
        Object.prototype.hasOwnProperty.call(t, i) && (r[i] = t[i]);
    }
    return r;
  }, se.apply(this, arguments);
};
function Ae(e, r, t) {
  if (t || arguments.length === 2)
    for (var n = 0, a = r.length, i; n < a; n++)
      (i || !(n in r)) && (i || (i = Array.prototype.slice.call(r, 0, n)), i[n] = r[n]);
  return e.concat(i || Array.prototype.slice.call(r));
}
function Zn(e) {
  var r = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return r[t] === void 0 && (r[t] = e(t)), r[t];
  };
}
var Qn = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, eo = /* @__PURE__ */ Zn(
  function(e) {
    return Qn.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
), X = "-ms-", He = "-moz-", U = "-webkit-", zt = "comm", sr = "rule", Br = "decl", ro = "@import", Vt = "@keyframes", to = "@layer", no = Math.abs, Ur = String.fromCharCode, jr = Object.assign;
function oo(e, r) {
  return ee(e, 0) ^ 45 ? (((r << 2 ^ ee(e, 0)) << 2 ^ ee(e, 1)) << 2 ^ ee(e, 2)) << 2 ^ ee(e, 3) : 0;
}
function qt(e) {
  return e.trim();
}
function Ee(e, r) {
  return (e = r.exec(e)) ? e[0] : e;
}
function F(e, r, t) {
  return e.replace(r, t);
}
function rr(e, r) {
  return e.indexOf(r);
}
function ee(e, r) {
  return e.charCodeAt(r) | 0;
}
function Me(e, r, t) {
  return e.slice(r, t);
}
function ge(e) {
  return e.length;
}
function Bt(e) {
  return e.length;
}
function Ge(e, r) {
  return r.push(e), e;
}
function ao(e, r) {
  return e.map(r).join("");
}
function St(e, r) {
  return e.filter(function(t) {
    return !Ee(t, r);
  });
}
var cr = 1, Fe = 1, Ut = 0, he = 0, Q = 0, ze = "";
function ur(e, r, t, n, a, i, s, f) {
  return { value: e, root: r, parent: t, type: n, props: a, children: i, line: cr, column: Fe, length: s, return: "", siblings: f };
}
function _e(e, r) {
  return jr(ur("", null, null, "", null, null, 0, e.siblings), e, { length: -e.length }, r);
}
function Ne(e) {
  for (; e.root; )
    e = _e(e.root, { children: [e] });
  Ge(e, e.siblings);
}
function io() {
  return Q;
}
function so() {
  return Q = he > 0 ? ee(ze, --he) : 0, Fe--, Q === 10 && (Fe = 1, cr--), Q;
}
function ve() {
  return Q = he < Ut ? ee(ze, he++) : 0, Fe++, Q === 10 && (Fe = 1, cr++), Q;
}
function Pe() {
  return ee(ze, he);
}
function tr() {
  return he;
}
function lr(e, r) {
  return Me(ze, e, r);
}
function Ir(e) {
  switch (e) {
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    case 59:
    case 123:
    case 125:
      return 4;
    case 58:
      return 3;
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function co(e) {
  return cr = Fe = 1, Ut = ge(ze = e), he = 0, [];
}
function uo(e) {
  return ze = "", e;
}
function _r(e) {
  return qt(lr(he - 1, $r(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function lo(e) {
  for (; (Q = Pe()) && Q < 33; )
    ve();
  return Ir(e) > 2 || Ir(Q) > 3 ? "" : " ";
}
function fo(e, r) {
  for (; --r && ve() && !(Q < 48 || Q > 102 || Q > 57 && Q < 65 || Q > 70 && Q < 97); )
    ;
  return lr(e, tr() + (r < 6 && Pe() == 32 && ve() == 32));
}
function $r(e) {
  for (; ve(); )
    switch (Q) {
      case e:
        return he;
      case 34:
      case 39:
        e !== 34 && e !== 39 && $r(Q);
        break;
      case 40:
        e === 41 && $r(e);
        break;
      case 92:
        ve();
        break;
    }
  return he;
}
function po(e, r) {
  for (; ve() && e + Q !== 47 + 10; )
    if (e + Q === 42 + 42 && Pe() === 47)
      break;
  return "/*" + lr(r, he - 1) + "*" + Ur(e === 47 ? e : ve());
}
function ho(e) {
  for (; !Ir(Pe()); )
    ve();
  return lr(e, he);
}
function yo(e) {
  return uo(nr("", null, null, null, [""], e = co(e), 0, [0], e));
}
function nr(e, r, t, n, a, i, s, f, l) {
  for (var g = 0, p = 0, y = s, b = 0, w = 0, j = 0, _ = 1, W = 1, S = 1, E = 0, T = "", P = a, d = i, I = n, k = T; W; )
    switch (j = E, E = ve()) {
      case 40:
        if (j != 108 && ee(k, y - 1) == 58) {
          rr(k += F(_r(E), "&", "&\f"), "&\f") != -1 && (S = -1);
          break;
        }
      case 34:
      case 39:
      case 91:
        k += _r(E);
        break;
      case 9:
      case 10:
      case 13:
      case 32:
        k += lo(j);
        break;
      case 92:
        k += fo(tr() - 1, 7);
        continue;
      case 47:
        switch (Pe()) {
          case 42:
          case 47:
            Ge(vo(po(ve(), tr()), r, t, l), l);
            break;
          default:
            k += "/";
        }
        break;
      case 123 * _:
        f[g++] = ge(k) * S;
      case 125 * _:
      case 59:
      case 0:
        switch (E) {
          case 0:
          case 125:
            W = 0;
          case 59 + p:
            S == -1 && (k = F(k, /\f/g, "")), w > 0 && ge(k) - y && Ge(w > 32 ? Tt(k + ";", n, t, y - 1, l) : Tt(F(k, " ", "") + ";", n, t, y - 2, l), l);
            break;
          case 59:
            k += ";";
          default:
            if (Ge(I = _t(k, r, t, g, p, a, f, T, P = [], d = [], y, i), i), E === 123)
              if (p === 0)
                nr(k, r, I, I, P, i, y, f, d);
              else
                switch (b === 99 && ee(k, 3) === 110 ? 100 : b) {
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    nr(e, I, I, n && Ge(_t(e, I, I, 0, 0, a, f, T, a, P = [], y, d), d), a, d, y, f, n ? P : d);
                    break;
                  default:
                    nr(k, I, I, I, [""], d, 0, f, d);
                }
        }
        g = p = w = 0, _ = S = 1, T = k = "", y = s;
        break;
      case 58:
        y = 1 + ge(k), w = j;
      default:
        if (_ < 1) {
          if (E == 123)
            --_;
          else if (E == 125 && _++ == 0 && so() == 125)
            continue;
        }
        switch (k += Ur(E), E * _) {
          case 38:
            S = p > 0 ? 1 : (k += "\f", -1);
            break;
          case 44:
            f[g++] = (ge(k) - 1) * S, S = 1;
            break;
          case 64:
            Pe() === 45 && (k += _r(ve())), b = Pe(), p = y = ge(T = k += ho(tr())), E++;
            break;
          case 45:
            j === 45 && ge(k) == 2 && (_ = 0);
        }
    }
  return i;
}
function _t(e, r, t, n, a, i, s, f, l, g, p, y) {
  for (var b = a - 1, w = a === 0 ? i : [""], j = Bt(w), _ = 0, W = 0, S = 0; _ < n; ++_)
    for (var E = 0, T = Me(e, b + 1, b = no(W = s[_])), P = e; E < j; ++E)
      (P = qt(W > 0 ? w[E] + " " + T : F(T, /&\f/g, w[E]))) && (l[S++] = P);
  return ur(e, r, t, a === 0 ? sr : f, l, g, p, y);
}
function vo(e, r, t, n) {
  return ur(e, r, t, zt, Ur(io()), Me(e, 2, -2), 0, n);
}
function Tt(e, r, t, n, a) {
  return ur(e, r, t, Br, Me(e, 0, n), Me(e, n + 1, -1), n, a);
}
function Gt(e, r, t) {
  switch (oo(e, r)) {
    case 5103:
      return U + "print-" + e + e;
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return U + e + e;
    case 4789:
      return He + e + e;
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return U + e + He + e + X + e + e;
    case 5936:
      switch (ee(e, r + 11)) {
        case 114:
          return U + e + X + F(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        case 108:
          return U + e + X + F(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        case 45:
          return U + e + X + F(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
    case 6828:
    case 4268:
    case 2903:
      return U + e + X + e + e;
    case 6165:
      return U + e + X + "flex-" + e + e;
    case 5187:
      return U + e + F(e, /(\w+).+(:[^]+)/, U + "box-$1$2" + X + "flex-$1$2") + e;
    case 5443:
      return U + e + X + "flex-item-" + F(e, /flex-|-self/g, "") + (Ee(e, /flex-|baseline/) ? "" : X + "grid-row-" + F(e, /flex-|-self/g, "")) + e;
    case 4675:
      return U + e + X + "flex-line-pack" + F(e, /align-content|flex-|-self/g, "") + e;
    case 5548:
      return U + e + X + F(e, "shrink", "negative") + e;
    case 5292:
      return U + e + X + F(e, "basis", "preferred-size") + e;
    case 6060:
      return U + "box-" + F(e, "-grow", "") + U + e + X + F(e, "grow", "positive") + e;
    case 4554:
      return U + F(e, /([^-])(transform)/g, "$1" + U + "$2") + e;
    case 6187:
      return F(F(F(e, /(zoom-|grab)/, U + "$1"), /(image-set)/, U + "$1"), e, "") + e;
    case 5495:
    case 3959:
      return F(e, /(image-set\([^]*)/, U + "$1$`$1");
    case 4968:
      return F(F(e, /(.+:)(flex-)?(.*)/, U + "box-pack:$3" + X + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + U + e + e;
    case 4200:
      if (!Ee(e, /flex-|baseline/))
        return X + "grid-column-align" + Me(e, r) + e;
      break;
    case 2592:
    case 3360:
      return X + F(e, "template-", "") + e;
    case 4384:
    case 3616:
      return t && t.some(function(n, a) {
        return r = a, Ee(n.props, /grid-\w+-end/);
      }) ? ~rr(e + (t = t[r].value), "span") ? e : X + F(e, "-start", "") + e + X + "grid-row-span:" + (~rr(t, "span") ? Ee(t, /\d+/) : +Ee(t, /\d+/) - +Ee(e, /\d+/)) + ";" : X + F(e, "-start", "") + e;
    case 4896:
    case 4128:
      return t && t.some(function(n) {
        return Ee(n.props, /grid-\w+-start/);
      }) ? e : X + F(F(e, "-end", "-span"), "span ", "") + e;
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return F(e, /(.+)-inline(.+)/, U + "$1$2") + e;
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (ge(e) - 1 - r > 6)
        switch (ee(e, r + 1)) {
          case 109:
            if (ee(e, r + 4) !== 45)
              break;
          case 102:
            return F(e, /(.+:)(.+)-([^]+)/, "$1" + U + "$2-$3$1" + He + (ee(e, r + 3) == 108 ? "$3" : "$2-$3")) + e;
          case 115:
            return ~rr(e, "stretch") ? Gt(F(e, "stretch", "fill-available"), r, t) + e : e;
        }
      break;
    case 5152:
    case 5920:
      return F(e, /(.+?):(\d+)(\s*\/\s*(span)?\s*(\d+))?(.*)/, function(n, a, i, s, f, l, g) {
        return X + a + ":" + i + g + (s ? X + a + "-span:" + (f ? l : +l - +i) + g : "") + e;
      });
    case 4949:
      if (ee(e, r + 6) === 121)
        return F(e, ":", ":" + U) + e;
      break;
    case 6444:
      switch (ee(e, ee(e, 14) === 45 ? 18 : 11)) {
        case 120:
          return F(e, /(.+:)([^;\s!]+)(;|(\s+)?!.+)?/, "$1" + U + (ee(e, 14) === 45 ? "inline-" : "") + "box$3$1" + U + "$2$3$1" + X + "$2box$3") + e;
        case 100:
          return F(e, ":", ":" + X) + e;
      }
      break;
    case 5719:
    case 2647:
    case 2135:
    case 3927:
    case 2391:
      return F(e, "scroll-", "scroll-snap-") + e;
  }
  return e;
}
function ar(e, r) {
  for (var t = "", n = 0; n < e.length; n++)
    t += r(e[n], n, e, r) || "";
  return t;
}
function mo(e, r, t, n) {
  switch (e.type) {
    case to:
      if (e.children.length)
        break;
    case ro:
    case Br:
      return e.return = e.return || e.value;
    case zt:
      return "";
    case Vt:
      return e.return = e.value + "{" + ar(e.children, n) + "}";
    case sr:
      if (!ge(e.value = e.props.join(",")))
        return "";
  }
  return ge(t = ar(e.children, n)) ? e.return = e.value + "{" + t + "}" : "";
}
function go(e) {
  var r = Bt(e);
  return function(t, n, a, i) {
    for (var s = "", f = 0; f < r; f++)
      s += e[f](t, n, a, i) || "";
    return s;
  };
}
function bo(e) {
  return function(r) {
    r.root || (r = r.return) && e(r);
  };
}
function Eo(e, r, t, n) {
  if (e.length > -1 && !e.return)
    switch (e.type) {
      case Br:
        e.return = Gt(e.value, e.length, t);
        return;
      case Vt:
        return ar([_e(e, { value: F(e.value, "@", "@" + U) })], n);
      case sr:
        if (e.length)
          return ao(t = e.props, function(a) {
            switch (Ee(a, n = /(::plac\w+|:read-\w+)/)) {
              case ":read-only":
              case ":read-write":
                Ne(_e(e, { props: [F(a, /:(read-\w+)/, ":" + He + "$1")] })), Ne(_e(e, { props: [a] })), jr(e, { props: St(t, n) });
                break;
              case "::placeholder":
                Ne(_e(e, { props: [F(a, /:(plac\w+)/, ":" + U + "input-$1")] })), Ne(_e(e, { props: [F(a, /:(plac\w+)/, ":" + He + "$1")] })), Ne(_e(e, { props: [F(a, /:(plac\w+)/, X + "input-$1")] })), Ne(_e(e, { props: [a] })), jr(e, { props: St(t, n) });
                break;
            }
            return "";
          });
    }
}
var wo = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
}, je = typeof process < "u" && process.env !== void 0 && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", Gr = typeof window < "u" && "HTMLElement" in window, So = !!(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), Rt = /invalid hook call/i, Ze = /* @__PURE__ */ new Set(), _o = function(e, r) {
  if (process.env.NODE_ENV !== "production") {
    var t = r ? ' with the id of "'.concat(r, '"') : "", n = "The component ".concat(e).concat(t, ` has been created dynamically.
`) + `You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.`, a = console.error;
    try {
      var i = !0;
      console.error = function(s) {
        for (var f = [], l = 1; l < arguments.length; l++)
          f[l - 1] = arguments[l];
        Rt.test(s) ? (i = !1, Ze.delete(n)) : a.apply(void 0, Ae([s], f, !1));
      }, Lt(), i && !Ze.has(n) && (console.warn(n), Ze.add(n));
    } catch (s) {
      Rt.test(s.message) && Ze.delete(n);
    } finally {
      console.error = a;
    }
  }
}, fr = Object.freeze([]), Le = Object.freeze({});
function To(e, r, t) {
  return t === void 0 && (t = Le), e.theme !== t.theme && e.theme || r || t.theme;
}
var Dr = /* @__PURE__ */ new Set(["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "tr", "track", "u", "ul", "use", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "tspan"]), Ro = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, xo = /(^-|-$)/g;
function xt(e) {
  return e.replace(Ro, "-").replace(xo, "");
}
var Co = /(a)(d)/gi, Ct = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function Nr(e) {
  var r, t = "";
  for (r = Math.abs(e); r > 52; r = r / 52 | 0)
    t = Ct(r % 52) + t;
  return (Ct(r % 52) + t).replace(Co, "$1-$2");
}
var Tr, Ce = function(e, r) {
  for (var t = r.length; t; )
    e = 33 * e ^ r.charCodeAt(--t);
  return e;
}, Ht = function(e) {
  return Ce(5381, e);
};
function Xt(e) {
  return Nr(Ht(e) >>> 0);
}
function Kt(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function Rr(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var Jt = typeof Symbol == "function" && Symbol.for, Zt = Jt ? Symbol.for("react.memo") : 60115, Oo = Jt ? Symbol.for("react.forward_ref") : 60112, Po = { childContextTypes: !0, contextType: !0, contextTypes: !0, defaultProps: !0, displayName: !0, getDefaultProps: !0, getDerivedStateFromError: !0, getDerivedStateFromProps: !0, mixins: !0, propTypes: !0, type: !0 }, ko = { name: !0, length: !0, prototype: !0, caller: !0, callee: !0, arguments: !0, arity: !0 }, Qt = { $$typeof: !0, compare: !0, defaultProps: !0, displayName: !0, propTypes: !0, type: !0 }, Ao = ((Tr = {})[Oo] = { $$typeof: !0, render: !0, defaultProps: !0, displayName: !0, propTypes: !0 }, Tr[Zt] = Qt, Tr);
function Ot(e) {
  return ("type" in (r = e) && r.type.$$typeof) === Zt ? Qt : "$$typeof" in e ? Ao[e.$$typeof] : Po;
  var r;
}
var jo = Object.defineProperty, Io = Object.getOwnPropertyNames, Pt = Object.getOwnPropertySymbols, $o = Object.getOwnPropertyDescriptor, Do = Object.getPrototypeOf, kt = Object.prototype;
function en(e, r, t) {
  if (typeof r != "string") {
    if (kt) {
      var n = Do(r);
      n && n !== kt && en(e, n, t);
    }
    var a = Io(r);
    Pt && (a = a.concat(Pt(r)));
    for (var i = Ot(e), s = Ot(r), f = 0; f < a.length; ++f) {
      var l = a[f];
      if (!(l in ko || t && t[l] || s && l in s || i && l in i)) {
        var g = $o(r, l);
        try {
          jo(e, l, g);
        } catch {
        }
      }
    }
  }
  return e;
}
function Ye(e) {
  return typeof e == "function";
}
function Hr(e) {
  return typeof e == "object" && "styledComponentId" in e;
}
function Oe(e, r) {
  return e && r ? "".concat(e, " ").concat(r) : e || r || "";
}
function Mr(e, r) {
  if (e.length === 0)
    return "";
  for (var t = e[0], n = 1; n < e.length; n++)
    t += r ? r + e[n] : e[n];
  return t;
}
function We(e) {
  return e !== null && typeof e == "object" && e.constructor.name === Object.name && !("props" in e && e.$$typeof);
}
function Fr(e, r, t) {
  if (t === void 0 && (t = !1), !t && !We(e) && !Array.isArray(e))
    return r;
  if (Array.isArray(r))
    for (var n = 0; n < r.length; n++)
      e[n] = Fr(e[n], r[n]);
  else if (We(r))
    for (var n in r)
      e[n] = Fr(e[n], r[n]);
  return e;
}
function Xr(e, r) {
  Object.defineProperty(e, "toString", { value: r });
}
var No = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
`, 18: "ThemeProvider: Please make sure your useTheme hook is within a `<ThemeProvider>`" } : {};
function Mo() {
  for (var e = [], r = 0; r < arguments.length; r++)
    e[r] = arguments[r];
  for (var t = e[0], n = [], a = 1, i = e.length; a < i; a += 1)
    n.push(e[a]);
  return n.forEach(function(s) {
    t = t.replace(/%[a-z]/, s);
  }), t;
}
function Ve(e) {
  for (var r = [], t = 1; t < arguments.length; t++)
    r[t - 1] = arguments[t];
  return process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#".concat(e, " for more information.").concat(r.length > 0 ? " Args: ".concat(r.join(", ")) : "")) : new Error(Mo.apply(void 0, Ae([No[e]], r, !1)).trim());
}
var Fo = function() {
  function e(r) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = r;
  }
  return e.prototype.indexOfGroup = function(r) {
    for (var t = 0, n = 0; n < r; n++)
      t += this.groupSizes[n];
    return t;
  }, e.prototype.insertRules = function(r, t) {
    if (r >= this.groupSizes.length) {
      for (var n = this.groupSizes, a = n.length, i = a; r >= i; )
        if ((i <<= 1) < 0)
          throw Ve(16, "".concat(r));
      this.groupSizes = new Uint32Array(i), this.groupSizes.set(n), this.length = i;
      for (var s = a; s < i; s++)
        this.groupSizes[s] = 0;
    }
    for (var f = this.indexOfGroup(r + 1), l = (s = 0, t.length); s < l; s++)
      this.tag.insertRule(f, t[s]) && (this.groupSizes[r]++, f++);
  }, e.prototype.clearGroup = function(r) {
    if (r < this.length) {
      var t = this.groupSizes[r], n = this.indexOfGroup(r), a = n + t;
      this.groupSizes[r] = 0;
      for (var i = n; i < a; i++)
        this.tag.deleteRule(n);
    }
  }, e.prototype.getGroup = function(r) {
    var t = "";
    if (r >= this.length || this.groupSizes[r] === 0)
      return t;
    for (var n = this.groupSizes[r], a = this.indexOfGroup(r), i = a + n, s = a; s < i; s++)
      t += "".concat(this.tag.getRule(s)).concat(`/*!sc*/
`);
    return t;
  }, e;
}(), or = /* @__PURE__ */ new Map(), ir = /* @__PURE__ */ new Map(), xr = 1, Qe = function(e) {
  if (or.has(e))
    return or.get(e);
  for (; ir.has(xr); )
    xr++;
  var r = xr++;
  if (process.env.NODE_ENV !== "production" && ((0 | r) < 0 || r > 1073741824))
    throw Ve(16, "".concat(r));
  return or.set(e, r), ir.set(r, e), r;
}, Lo = function(e, r) {
  or.set(e, r), ir.set(r, e);
}, Yo = "style[".concat(je, "][").concat("data-styled-version", '="').concat("6.0.8", '"]'), Wo = new RegExp("^".concat(je, '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)')), zo = function(e, r, t) {
  for (var n, a = t.split(","), i = 0, s = a.length; i < s; i++)
    (n = a[i]) && e.registerName(r, n);
}, Vo = function(e, r) {
  for (var t, n = ((t = r.textContent) !== null && t !== void 0 ? t : "").split(`/*!sc*/
`), a = [], i = 0, s = n.length; i < s; i++) {
    var f = n[i].trim();
    if (f) {
      var l = f.match(Wo);
      if (l) {
        var g = 0 | parseInt(l[1], 10), p = l[2];
        g !== 0 && (Lo(p, g), zo(e, p, l[3]), e.getTag().insertRules(g, a)), a.length = 0;
      } else
        a.push(f);
    }
  }
};
function qo() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}
var rn = function(e) {
  var r = document.head, t = e || r, n = document.createElement("style"), a = function(f) {
    var l = Array.from(f.querySelectorAll("style[".concat(je, "]")));
    return l[l.length - 1];
  }(t), i = a !== void 0 ? a.nextSibling : null;
  n.setAttribute(je, "active"), n.setAttribute("data-styled-version", "6.0.8");
  var s = qo();
  return s && n.setAttribute("nonce", s), t.insertBefore(n, i), n;
}, Bo = function() {
  function e(r) {
    this.element = rn(r), this.element.appendChild(document.createTextNode("")), this.sheet = function(t) {
      if (t.sheet)
        return t.sheet;
      for (var n = document.styleSheets, a = 0, i = n.length; a < i; a++) {
        var s = n[a];
        if (s.ownerNode === t)
          return s;
      }
      throw Ve(17);
    }(this.element), this.length = 0;
  }
  return e.prototype.insertRule = function(r, t) {
    try {
      return this.sheet.insertRule(t, r), this.length++, !0;
    } catch {
      return !1;
    }
  }, e.prototype.deleteRule = function(r) {
    this.sheet.deleteRule(r), this.length--;
  }, e.prototype.getRule = function(r) {
    var t = this.sheet.cssRules[r];
    return t && t.cssText ? t.cssText : "";
  }, e;
}(), Uo = function() {
  function e(r) {
    this.element = rn(r), this.nodes = this.element.childNodes, this.length = 0;
  }
  return e.prototype.insertRule = function(r, t) {
    if (r <= this.length && r >= 0) {
      var n = document.createTextNode(t);
      return this.element.insertBefore(n, this.nodes[r] || null), this.length++, !0;
    }
    return !1;
  }, e.prototype.deleteRule = function(r) {
    this.element.removeChild(this.nodes[r]), this.length--;
  }, e.prototype.getRule = function(r) {
    return r < this.length ? this.nodes[r].textContent : "";
  }, e;
}(), Go = function() {
  function e(r) {
    this.rules = [], this.length = 0;
  }
  return e.prototype.insertRule = function(r, t) {
    return r <= this.length && (this.rules.splice(r, 0, t), this.length++, !0);
  }, e.prototype.deleteRule = function(r) {
    this.rules.splice(r, 1), this.length--;
  }, e.prototype.getRule = function(r) {
    return r < this.length ? this.rules[r] : "";
  }, e;
}(), At = Gr, Ho = { isServer: !Gr, useCSSOMInjection: !So }, tn = function() {
  function e(r, t, n) {
    r === void 0 && (r = Le), t === void 0 && (t = {});
    var a = this;
    this.options = se(se({}, Ho), r), this.gs = t, this.names = new Map(n), this.server = !!r.isServer, !this.server && Gr && At && (At = !1, function(i) {
      for (var s = document.querySelectorAll(Yo), f = 0, l = s.length; f < l; f++) {
        var g = s[f];
        g && g.getAttribute(je) !== "active" && (Vo(i, g), g.parentNode && g.parentNode.removeChild(g));
      }
    }(this)), Xr(this, function() {
      return function(i) {
        for (var s = i.getTag(), f = s.length, l = "", g = function(y) {
          var b = function(S) {
            return ir.get(S);
          }(y);
          if (b === void 0)
            return "continue";
          var w = i.names.get(b), j = s.getGroup(y);
          if (w === void 0 || j.length === 0)
            return "continue";
          var _ = "".concat(je, ".g").concat(y, '[id="').concat(b, '"]'), W = "";
          w !== void 0 && w.forEach(function(S) {
            S.length > 0 && (W += "".concat(S, ","));
          }), l += "".concat(j).concat(_, '{content:"').concat(W, '"}').concat(`/*!sc*/
`);
        }, p = 0; p < f; p++)
          g(p);
        return l;
      }(a);
    });
  }
  return e.registerId = function(r) {
    return Qe(r);
  }, e.prototype.reconstructWithOptions = function(r, t) {
    return t === void 0 && (t = !0), new e(se(se({}, this.options), r), this.gs, t && this.names || void 0);
  }, e.prototype.allocateGSInstance = function(r) {
    return this.gs[r] = (this.gs[r] || 0) + 1;
  }, e.prototype.getTag = function() {
    return this.tag || (this.tag = (r = function(t) {
      var n = t.useCSSOMInjection, a = t.target;
      return t.isServer ? new Go(a) : n ? new Bo(a) : new Uo(a);
    }(this.options), new Fo(r)));
    var r;
  }, e.prototype.hasNameForId = function(r, t) {
    return this.names.has(r) && this.names.get(r).has(t);
  }, e.prototype.registerName = function(r, t) {
    if (Qe(r), this.names.has(r))
      this.names.get(r).add(t);
    else {
      var n = /* @__PURE__ */ new Set();
      n.add(t), this.names.set(r, n);
    }
  }, e.prototype.insertRules = function(r, t, n) {
    this.registerName(r, t), this.getTag().insertRules(Qe(r), n);
  }, e.prototype.clearNames = function(r) {
    this.names.has(r) && this.names.get(r).clear();
  }, e.prototype.clearRules = function(r) {
    this.getTag().clearGroup(Qe(r)), this.clearNames(r);
  }, e.prototype.clearTag = function() {
    this.tag = void 0;
  }, e;
}(), Xo = /&/g, Ko = /^\s*\/\/.*$/gm;
function nn(e, r) {
  return e.map(function(t) {
    return t.type === "rule" && (t.value = "".concat(r, " ").concat(t.value), t.value = t.value.replaceAll(",", ",".concat(r, " ")), t.props = t.props.map(function(n) {
      return "".concat(r, " ").concat(n);
    })), Array.isArray(t.children) && t.type !== "@keyframes" && (t.children = nn(t.children, r)), t;
  });
}
function Jo(e) {
  var r, t, n, a = e === void 0 ? Le : e, i = a.options, s = i === void 0 ? Le : i, f = a.plugins, l = f === void 0 ? fr : f, g = function(b, w, j) {
    return j === t || j.startsWith(t) && j.endsWith(t) && j.replaceAll(t, "").length > 0 ? ".".concat(r) : b;
  }, p = l.slice();
  p.push(function(b) {
    b.type === sr && b.value.includes("&") && (b.props[0] = b.props[0].replace(Xo, t).replace(n, g));
  }), s.prefix && p.push(Eo), p.push(mo);
  var y = function(b, w, j, _) {
    w === void 0 && (w = ""), j === void 0 && (j = ""), _ === void 0 && (_ = "&"), r = _, t = w, n = new RegExp("\\".concat(t, "\\b"), "g");
    var W = b.replace(Ko, ""), S = yo(j || w ? "".concat(j, " ").concat(w, " { ").concat(W, " }") : W);
    s.namespace && (S = nn(S, s.namespace));
    var E = [];
    return ar(S, go(p.concat(bo(function(T) {
      return E.push(T);
    })))), E;
  };
  return y.hash = l.length ? l.reduce(function(b, w) {
    return w.name || Ve(15), Ce(b, w.name);
  }, 5381).toString() : "", y;
}
var Zo = new tn(), Lr = Jo(), on = me.createContext({ shouldForwardProp: void 0, styleSheet: Zo, stylis: Lr });
on.Consumer;
me.createContext(void 0);
function jt() {
  return jn(on);
}
var Yr = function() {
  function e(r, t) {
    var n = this;
    this.inject = function(a, i) {
      i === void 0 && (i = Lr);
      var s = n.name + i.hash;
      a.hasNameForId(n.id, s) || a.insertRules(n.id, s, i(n.rules, s, "@keyframes"));
    }, this.name = r, this.id = "sc-keyframes-".concat(r), this.rules = t, Xr(this, function() {
      throw Ve(12, String(n.name));
    });
  }
  return e.prototype.getName = function(r) {
    return r === void 0 && (r = Lr), this.name + r.hash;
  }, e;
}(), Qo = function(e) {
  return e >= "A" && e <= "Z";
};
function It(e) {
  for (var r = "", t = 0; t < e.length; t++) {
    var n = e[t];
    if (t === 1 && n === "-" && e[0] === "-")
      return e;
    Qo(n) ? r += "-" + n.toLowerCase() : r += n;
  }
  return r.startsWith("ms-") ? "-" + r : r;
}
var an = function(e) {
  return e == null || e === !1 || e === "";
}, sn = function(e) {
  var r, t, n = [];
  for (var a in e) {
    var i = e[a];
    e.hasOwnProperty(a) && !an(i) && (Array.isArray(i) && i.isCss || Ye(i) ? n.push("".concat(It(a), ":"), i, ";") : We(i) ? n.push.apply(n, Ae(Ae(["".concat(a, " {")], sn(i), !1), ["}"], !1)) : n.push("".concat(It(a), ": ").concat((r = a, (t = i) == null || typeof t == "boolean" || t === "" ? "" : typeof t != "number" || t === 0 || r in wo || r.startsWith("--") ? String(t).trim() : "".concat(t, "px")), ";")));
  }
  return n;
};
function ke(e, r, t, n) {
  if (an(e))
    return [];
  if (Hr(e))
    return [".".concat(e.styledComponentId)];
  if (Ye(e)) {
    if (!Ye(i = e) || i.prototype && i.prototype.isReactComponent || !r)
      return [e];
    var a = e(r);
    return process.env.NODE_ENV === "production" || typeof a != "object" || Array.isArray(a) || a instanceof Yr || We(a) || a === null || console.error("".concat(Kt(e), " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.")), ke(a, r, t, n);
  }
  var i;
  return e instanceof Yr ? t ? (e.inject(t, n), [e.getName(n)]) : [e] : We(e) ? sn(e) : Array.isArray(e) ? Array.prototype.concat.apply(fr, e.map(function(s) {
    return ke(s, r, t, n);
  })) : [e.toString()];
}
function ea(e) {
  for (var r = 0; r < e.length; r += 1) {
    var t = e[r];
    if (Ye(t) && !Hr(t))
      return !1;
  }
  return !0;
}
var ra = Ht("6.0.8"), ta = function() {
  function e(r, t, n) {
    this.rules = r, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (n === void 0 || n.isStatic) && ea(r), this.componentId = t, this.baseHash = Ce(ra, t), this.baseStyle = n, tn.registerId(t);
  }
  return e.prototype.generateAndInjectStyles = function(r, t, n) {
    var a = this.baseStyle ? this.baseStyle.generateAndInjectStyles(r, t, n) : "";
    if (this.isStatic && !n.hash)
      if (this.staticRulesId && t.hasNameForId(this.componentId, this.staticRulesId))
        a = Oe(a, this.staticRulesId);
      else {
        var i = Mr(ke(this.rules, r, t, n)), s = Nr(Ce(this.baseHash, i) >>> 0);
        if (!t.hasNameForId(this.componentId, s)) {
          var f = n(i, ".".concat(s), void 0, this.componentId);
          t.insertRules(this.componentId, s, f);
        }
        a = Oe(a, s), this.staticRulesId = s;
      }
    else {
      for (var l = Ce(this.baseHash, n.hash), g = "", p = 0; p < this.rules.length; p++) {
        var y = this.rules[p];
        if (typeof y == "string")
          g += y, process.env.NODE_ENV !== "production" && (l = Ce(l, y));
        else if (y) {
          var b = Mr(ke(y, r, t, n));
          l = Ce(l, b + p), g += b;
        }
      }
      if (g) {
        var w = Nr(l >>> 0);
        t.hasNameForId(this.componentId, w) || t.insertRules(this.componentId, w, n(g, ".".concat(w), void 0, this.componentId)), a = Oe(a, w);
      }
    }
    return a;
  }, e;
}(), cn = me.createContext(void 0);
cn.Consumer;
var Cr = {}, $t = /* @__PURE__ */ new Set();
function na(e, r, t) {
  var n = Hr(e), a = e, i = !Rr(e), s = r.attrs, f = s === void 0 ? fr : s, l = r.componentId, g = l === void 0 ? function(P, d) {
    var I = typeof P != "string" ? "sc" : xt(P);
    Cr[I] = (Cr[I] || 0) + 1;
    var k = "".concat(I, "-").concat(Xt("6.0.8" + I + Cr[I]));
    return d ? "".concat(d, "-").concat(k) : k;
  }(r.displayName, r.parentComponentId) : l, p = r.displayName, y = p === void 0 ? function(P) {
    return Rr(P) ? "styled.".concat(P) : "Styled(".concat(Kt(P), ")");
  }(e) : p, b = r.displayName && r.componentId ? "".concat(xt(r.displayName), "-").concat(r.componentId) : r.componentId || g, w = n && a.attrs ? a.attrs.concat(f).filter(Boolean) : f, j = r.shouldForwardProp;
  if (n && a.shouldForwardProp) {
    var _ = a.shouldForwardProp;
    if (r.shouldForwardProp) {
      var W = r.shouldForwardProp;
      j = function(P, d) {
        return _(P, d) && W(P, d);
      };
    } else
      j = _;
  }
  var S = new ta(t, b, n ? a.componentStyle : void 0);
  function E(P, d) {
    return function(I, k, ue) {
      var le = I.attrs, Re = I.componentStyle, we = I.defaultProps, Se = I.foldedComponentIds, re = I.styledComponentId, te = I.target, ye = me.useContext(cn), oe = jt(), ie = I.shouldForwardProp || oe.shouldForwardProp;
      process.env.NODE_ENV !== "production" && ut(re);
      var ne = function(R, M, C) {
        for (var x, A = se(se({}, M), { className: void 0, theme: C }), L = 0; L < R.length; L += 1) {
          var N = Ye(x = R[L]) ? x(A) : x;
          for (var H in N)
            A[H] = H === "className" ? Oe(A[H], N[H]) : H === "style" ? se(se({}, A[H]), N[H]) : N[H];
        }
        return M.className && (A.className = Oe(A.className, M.className)), A;
      }(le, k, To(k, ye, we) || Le), fe = ne.as || te, u = {};
      for (var h in ne)
        ne[h] === void 0 || h[0] === "$" || h === "as" || h === "theme" || (h === "forwardedAs" ? u.as = ne.forwardedAs : ie && !ie(h, fe) || (u[h] = ne[h], ie || process.env.NODE_ENV !== "development" || eo(h) || $t.has(h) || !Dr.has(fe) || ($t.add(h), console.warn('styled-components: it looks like an unknown prop "'.concat(h, '" is being sent through to the DOM, which will likely trigger a React console error. If you would like automatic filtering of unknown props, you can opt-into that behavior via `<StyleSheetManager shouldForwardProp={...}>` (connect an API like `@emotion/is-prop-valid`) or consider using transient props (`$` prefix for automatic filtering.)')))));
      var D = function(R, M) {
        var C = jt(), x = R.generateAndInjectStyles(M, C.styleSheet, C.stylis);
        return process.env.NODE_ENV !== "production" && ut(x), x;
      }(Re, ne);
      process.env.NODE_ENV !== "production" && I.warnTooManyClasses && I.warnTooManyClasses(D);
      var $ = Oe(Se, re);
      return D && ($ += " " + D), ne.className && ($ += " " + ne.className), u[Rr(fe) && !Dr.has(fe) ? "class" : "className"] = $, u.ref = ue, An(fe, u);
    }(T, P, d);
  }
  process.env.NODE_ENV !== "production" && (E.displayName = y);
  var T = me.forwardRef(E);
  return T.attrs = w, T.componentStyle = S, T.shouldForwardProp = j, process.env.NODE_ENV !== "production" && (T.displayName = y), T.foldedComponentIds = n ? Oe(a.foldedComponentIds, a.styledComponentId) : "", T.styledComponentId = b, T.target = n ? a.target : e, Object.defineProperty(T, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(P) {
    this._foldedDefaultProps = n ? function(d) {
      for (var I = [], k = 1; k < arguments.length; k++)
        I[k - 1] = arguments[k];
      for (var ue = 0, le = I; ue < le.length; ue++)
        Fr(d, le[ue], !0);
      return d;
    }({}, a.defaultProps, P) : P;
  } }), process.env.NODE_ENV !== "production" && (_o(y, b), T.warnTooManyClasses = function(P, d) {
    var I = {}, k = !1;
    return function(ue) {
      if (!k && (I[ue] = !0, Object.keys(I).length >= 200)) {
        var le = d ? ' with the id of "'.concat(d, '"') : "";
        console.warn("Over ".concat(200, " classes were generated for component ").concat(P).concat(le, `.
`) + `Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), k = !0, I = {};
      }
    };
  }(y, b)), Xr(T, function() {
    return ".".concat(T.styledComponentId);
  }), i && en(T, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0 }), T;
}
function Dt(e, r) {
  for (var t = [e[0]], n = 0, a = r.length; n < a; n += 1)
    t.push(r[n], e[n + 1]);
  return t;
}
var Nt = function(e) {
  return Object.assign(e, { isCss: !0 });
};
function un(e) {
  for (var r = [], t = 1; t < arguments.length; t++)
    r[t - 1] = arguments[t];
  if (Ye(e) || We(e)) {
    var n = e;
    return Nt(ke(Dt(fr, Ae([n], r, !0))));
  }
  var a = e;
  return r.length === 0 && a.length === 1 && typeof a[0] == "string" ? ke(a) : Nt(ke(Dt(a, r)));
}
function Wr(e, r, t) {
  if (t === void 0 && (t = Le), !r)
    throw Ve(1, r);
  var n = function(a) {
    for (var i = [], s = 1; s < arguments.length; s++)
      i[s - 1] = arguments[s];
    return e(r, t, un.apply(void 0, Ae([a], i, !1)));
  };
  return n.attrs = function(a) {
    return Wr(e, r, se(se({}, t), { attrs: Array.prototype.concat(t.attrs, a).filter(Boolean) }));
  }, n.withConfig = function(a) {
    return Wr(e, r, se(se({}, t), a));
  }, n;
}
var ln = function(e) {
  return Wr(na, e);
}, ce = ln;
Dr.forEach(function(e) {
  ce[e] = ln(e);
});
function oa(e) {
  for (var r = [], t = 1; t < arguments.length; t++)
    r[t - 1] = arguments[t];
  process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn("`keyframes` cannot be used on ReactNative, only on the web. To do animation in ReactNative please use Animated.");
  var n = Mr(un.apply(void 0, Ae([e], r, !1))), a = Xt(n);
  return new Yr(a, n);
}
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`);
var er = "__sc-".concat(je, "__");
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window[er] || (window[er] = 0), window[er] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window[er] += 1);
const aa = oa`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`, ia = ce.div`
  position: relative;
  max-width: ${(e) => e.carouselWidth || "20rem"};
  min-height: 10rem;
`, sa = ce.div`
  overflow: hidden;
  position: relative;
  min-height: 10rem;
`, ca = ce.div`
  display: flex;
  transition: transform 0.5s ease-out;
  aspect-ratio: 16 / 9;
  background-color: rgba(0, 0, 0, 0.3);
`, ua = ce.span`
  display: none;
  &.carousel-slide {
    opacity: 1;
    animation: ${aa} 1.2s ease-in;
  }

  &.carousel-slide.hidden {
    opacity: 0;
  }
`, la = ce.div`
  position: absolute;
  inset: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
`, Mt = ce.button`
  padding: 0.3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 100%;
  border: none;
  background-color: rgba(255, 255, 255, 0.4);
  color: #333;
  cursor: pointer;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: rgba(
      255,
      255,
      255,
      0.8
    ); /* Replace with your desired hover background color */
  }
  &:focus {
    border: none;
  }
`, fa = ce.div`
  position: absolute;
  bottom: 1rem;
  right: 0;
  left: 0;
  display: flex;
  justify-content: center;
`, da = ce.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
`, pa = ce.div`
  width: ${(e) => e.indicatorSize || "0.6rem"};
  height: ${(e) => e.indicatorSize || "0.6rem"};
  background-color: ${(e) => e.isActive ? "rgba(255, 255, 255, 0.9)" : "rgba(255, 255, 255, 0.8)"};
  border-radius: 50%;
  cursor: pointer;
  transition: background-color 0.3s ease;
  padding: ${(e) => e.isActive ? "0.2rem" : ""};
`, ha = ce.div`
  padding: 0.5rem;
  overflow-x: auto;
  display: flex;
  justify-content: start;
  max-width: 100%;
  background-color: #e2e2e27d !important;
  align-items: center;

  gap: 0.5rem;
  -ms-overflow-style: none; /* Hide scrollbar in IE and Edge */
  scrollbar-width: none; /* Hide scrollbar in Firefox */

  &::-webkit-scrollbar {
    width: 0.1em;
  }

  &::-webkit-scrollbar-thumb {
    background-color: transparent;
  }
`, ya = ce.div`
  cursor: pointer;
  opacity: ${(e) => e.isActive ? "1" : "0.6"};
  transition: opacity 0.3s ease;
  border: ${(e) => e.isActive ? "2px solid #333" : "none"};
  position: relative;
  img {
    width: ${(e) => e.thumbWidth};
  }
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0.1rem;
  padding-bottom: auto;
`, va = ce.span`
  min-width: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  img {
    width: 100%;
  }
`, ma = ce(ua)`
  min-width: 100%;
  display: ${(e) => e.isActive ? "block" : "none"};
  img {
    width: 100%;
  }
`, ba = ({
  slides: e = [],
  autoSlide: r = !1,
  autoSlideInterval: t = 2e3,
  carouselWidth: n = "20rem",
  thumbWidth: a = "3rem",
  showThumbs: i = !0,
  showControlArrow: s = !0,
  showIndicators: f = !0,
  controllArrowSize: l = 20,
  indicatorSize: g = "0.40rem",
  effect: p = "fade"
}) => {
  const [y, b] = In(0), w = Lt(null), j = () => {
    b((S) => (S + 1) % e.length);
  }, _ = () => {
    b(
      (S) => S === 0 ? e.length - 1 : S - 1
    );
  }, W = (S) => {
    b(S);
  };
  return lt(() => {
    if (!r)
      return;
    const S = setInterval(j, t);
    return () => clearInterval(S);
  }, []), lt(() => {
    w.current && w.current.scrollIntoView({
      behavior: "smooth",
      inline: "center"
    });
  }, [y]), /* @__PURE__ */ Z.jsx(Z.Fragment, { children: e.length > 0 && /* @__PURE__ */ Z.jsxs(ia, { carouselWidth: n, children: [
    /* @__PURE__ */ Z.jsxs(sa, { children: [
      /* @__PURE__ */ Z.jsxs(
        ca,
        {
          style: p === "fade" ? { transform: "" } : { transform: `translateX(-${y * 100}%)` },
          carouselWidth: n,
          children: [
            p === "fade" && e.map((S, E) => /* @__PURE__ */ Z.jsx(
              ma,
              {
                className: `carousel-slide ${y === E ? "active" : "hidden"}`,
                isActive: y === E,
                children: /* @__PURE__ */ Z.jsx("img", { src: S, alt: "" })
              },
              E
            )),
            p === "slide" && e.map((S, E) => /* @__PURE__ */ Z.jsx(va, { children: /* @__PURE__ */ Z.jsx("img", { src: S, alt: "" }) }, E))
          ]
        }
      ),
      s && e.length > 1 && /* @__PURE__ */ Z.jsxs(la, { children: [
        /* @__PURE__ */ Z.jsx(Mt, { onClick: _, children: /* @__PURE__ */ Z.jsx(Hn, { size: l }) }),
        /* @__PURE__ */ Z.jsx(Mt, { onClick: j, children: /* @__PURE__ */ Z.jsx(Jn, { size: l }) })
      ] }),
      f && e.length > 1 && /* @__PURE__ */ Z.jsx(fa, { children: /* @__PURE__ */ Z.jsx(da, { children: e.map((S, E) => /* @__PURE__ */ Z.jsx(
        pa,
        {
          indicatorSize: g,
          isActive: y === E,
          onClick: () => W(E)
        },
        E
      )) }) })
    ] }),
    i && e.length > 1 && /* @__PURE__ */ Z.jsx(ha, { children: e.map((S, E) => /* @__PURE__ */ Z.jsx(
      ya,
      {
        isActive: y === E,
        onClick: () => W(E),
        ref: (T) => {
          y === E && (w.current = T);
        },
        thumbWidth: a,
        children: /* @__PURE__ */ Z.jsx("img", { src: S, alt: "" })
      },
      E
    )) })
  ] }) });
};
export {
  ba as default
};
